AsyncDisplayKit/Private/ASEnvironmentInternal.h

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#import "ASEnvironment.h"
+
+#pragma once
+
+enum ASEnvironmentStatePropagation { DOWN, UP };
```

Comment:
Actually I wanted to use an enum class so it's strongly typed. Will change that to either an enum class or NS_ENUM